### ISSUE TITLE
Halotools

### ISSUE THREAD

**Issue by Lucia-Fonseca** [skypyproject/skypy#144]
## Description
[Halotools](https://halotools.readthedocs.io/en/latest/) Halotools is a specialized python package for building and testing models of the galaxy-halo connection, and analyzing catalogs of dark matter halos. 

The core functionality of the package includes:

- Fast generation of synthetic galaxy populations using HODs, abundance matching, and related methods
- Efficient algorithms for calculating galaxy clustering, lensing, z-space distortions, and other astronomical statistics
- A modular, object-oriented framework for designing your own galaxy evolution model

- End-to-end support for reducing halo catalogs and caching them as fast-loading hdf5 files

## References
- https://halotools.readthedocs.io/en/latest/

## Question
Do we feel we need this?

**Comment by bnord** [comment]
@Lucia-Fonseca would this perform simulation functions as the Vale+Ostriker implementation?

**Comment by gfavole** [comment]
@Lucia-Fonseca besides basic SHAM/HOD should we explore "decorated" models in which we match also secondary properties?

**Comment by bnord** [comment]
@gfavole what are some example secondary properties?

**Comment by Lucia-Fonseca** [comment]
That's a good question. It depends on whether that's within the scope of v0.2

**Comment by gfavole** [comment]
@bnord usually SHAM is performed using luminosity or stellar mass as main proxy for galaxies and maximum circular velocity for haloes. However, besides these primary properties, the best would be to match galaxies with haloes also in secondary props such as colours, age, halo assembly bias, etc.. An example of decorated SHAM is https://arxiv.org/pdf/1310.6747.pdf

**Comment by bnord** [comment]
Are these properties likely to be formatted in a table/list just like the primary properties? 
Or, put another away, do you see the data format requiring something to change for the secondary properties?

**Comment by gfavole** [comment]
yes, the secondary properties are formatted in a table exactly like the primary ones

**Comment by Lucia-Fonseca** [comment]
I do not see why these secondary properties should not be included

**Comment by bnord** [comment]
Yeah, if they're easy to generate now and then they are modular so that they can be changed later, I think we can go for it?

**Comment by bnord** [comment]
@Lucia-Fonseca Is it correct that halotools does NOT generate initial halo information, like mass distributions?

**Comment by gfavole** [comment]
I think yes, it doesn't generate those primary properties

**Comment by Lucia-Fonseca** [comment]
> @Lucia-Fonseca Is it correct that halotools does NOT generate initial halo information, like mass distributions?

Not an expert on Halotools. I think @gfavole knows more than me

**Comment by rrjbca** [comment]
Although it's good to think about secondary properties and plan ahead, I think we should prioritise getting a simple pipeline working for v0.2

**Comment by rrjbca** [comment]
Discussing with @aamara we should use halotools where possible as this is a requirement for astropy affiliation. It has [`HodModelFactory`](https://halotools.readthedocs.io/en/latest/api/halotools.empirical_models.HodModelFactory.html#) and [`SubhaloModelFactory`](https://halotools.readthedocs.io/en/latest/api/halotools.empirical_models.SubhaloModelFactory.html#) base classes that we can use to build our own HOD or SHAM style galaxy-halo connection models within their framework.
